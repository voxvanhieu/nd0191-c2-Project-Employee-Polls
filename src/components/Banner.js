import React from 'react';

export { Banner };

function Banner({ text }) {

    return (
        <div
            className='relative overflow-hidden flex items-center border border-default-200 dark:border-default-100 px-2 py-4 rounded-lg bg-gradient-to-r bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500 text-center mb-5'
            shadow="sm">
            <h2 data-testid="lblBanner" className="font-semibold text-xl text-foreground/90 w-full text-center">{text}</h2>
        </div>
    );
}