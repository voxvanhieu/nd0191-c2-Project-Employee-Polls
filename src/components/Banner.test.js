import React from 'react';
import { render, screen } from '@testing-library/react';
import { Banner } from './Banner';

describe('About page snapshot test', () => {
    it('will match snapshot', () => {
        const testText = "This is a sample text for testing";
        const component = render(
            <Banner text={testText} />
        );

        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();

        const lblBanner = screen.getByTestId("lblBanner");
        expect(lblBanner.textContent).toBe(testText);
    })
})
