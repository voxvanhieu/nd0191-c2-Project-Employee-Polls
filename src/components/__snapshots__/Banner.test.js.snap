// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page snapshot test will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative overflow-hidden flex items-center border border-default-200 dark:border-default-100 px-2 py-4 rounded-lg bg-gradient-to-r bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500 text-center mb-5"
        shadow="sm"
      >
        <h2
          class="font-semibold text-xl text-foreground/90 w-full text-center"
          data-testid="lblBanner"
        >
          This is a sample text for testing
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative overflow-hidden flex items-center border border-default-200 dark:border-default-100 px-2 py-4 rounded-lg bg-gradient-to-r bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500 text-center mb-5"
      shadow="sm"
    >
      <h2
        class="font-semibold text-xl text-foreground/90 w-full text-center"
        data-testid="lblBanner"
      >
        This is a sample text for testing
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
